src/cpp/transport/TCPTransportInterface.cpp

```diff
@@ -588,15 +450,22 @@ Locator_t TCPTransportInterface::RemoteToMainLocal(const Locator_t& remote) cons
 }
 
 
-bool TCPTransportInterface::CloseOutputChannel(const Locator_t& locator)
+void TCPTransportInterface::CloseOutputChannel(std::shared_ptr<TCPChannelResource>& channel)
 {
+    Locator_t physical_locator = channel->locator();
+    channel.reset();
```

Comment:
Why do you use a reference instead of a copy of the shared_ptr? It could delete the inner storage before you access the locator causing an invalid read. And you reset it without deleting it from the `channel_resources_` map, so you are resetting the shared_ptr contained in the map. Is that what you want to do? It will cause having invalid channel resources in the map.